AssemblerLib/MeshComponentMap.cpp

```diff
@@ -285,4 +285,39 @@ std::vector<GlobalIndexType> MeshComponentMap::getGlobalIndicesByComponent(
     return global_indices;
 }
 
+GlobalIndexType MeshComponentMap::getLocalIndex(
+    Location const& l,
+    std::size_t const comp_id,
+    std::size_t const range_begin,
+    std::size_t const range_end) const
+{
+    GlobalIndexType const global_index = getGlobalIndex(l, comp_id);
+#ifndef USE_PETSC
+    return global_index;
+#else
+    if (global_index >= 0)    // non-ghost location.
+        return global_index - range_begin;
+
+    //
+    // For a ghost location look up the global index in ghost indices.
+    //
+    GlobalIndexType index;
+
+    // A special case for a ghost location with global index equal to the size
+    // of the local vector:
+    if (-global_index == _num_global_dof) return 0;
+
+    auto const ghost_index_it = std::find(_ghosts_indices.begin(),
```

Comment:
Maybe it would be more efficient to implement it the other way around: Save local indices in the `_dict` and provide a method that converts local to global indices; since local indices are contiguous (are they?), they can be directly used as an index in a `ghost_index_local_to_global_index_mapping`.